### ISSUE TITLE
Reproducibility: Adding Metadata to Job description

### ISSUE THREAD

**Issue by bgoesswe** [Open-EO/openeo-api#195]
I think it would be a good idea to add the backend_version, which is active during the job execution, to the meta-data of the job. This adds transparency to the user and enables a better description of the execution context. 

Also related to #83: I suggest an optional meta-data entry for the input data PID related to an executed job, of course if the backend supports data citation.

I forked the "openeo-api" repository and made both suggested additions as an example here:
https://github.com/bgoesswe/openeo-api/tree/reproducibility 

I made the backend_version required, since it is anyways required in the "/" endpoint and it feels like not much effort to add it here too, but this may need some discussion.

**Comment by m-mohr** [comment]
For the next version, I'm planning to make a core API + extensions (sharing, subscriptions, ...) and I would also see reproducibility as a good extension. I don't have a workflow for adding extensions yet, but I'll come up with one after we have the next milestone done and then I'd be happy to see your suggestions for reproducibility added as an extension. Would that work for you?

**Comment by bgoesswe** [comment]
Yes sounds good, then I will prepare some ideas that we can discuss then.

**Comment by m-mohr** [comment]
Stale / solved by STAC?!
